### ISSUE TITLE
Missing Resources POST /api/resources/{id}/contracts at Swagger UI

### ISSUE THREAD

**Issue by jfernandezsqs** [International-Data-Spaces-Association/IDS-AppStore#47]
I have deployed the App Store and followed the example deatiled at (https://github.com/International-Data-Spaces-Association/IDS-AppStore/blob/main/scripts/tests/create_resources_and_upload_app_local.py) 
When I try to do lines 283 and 284 of the above file at the Swagger UI there is no POST /api/resources/{id}/contracts call to execute.

Is this a necessary call in order to have a complete resource at the App Store? so that the created Example App could be requested from another connector.

Could you please provide documentation explaining the necessary steps in order to create a complete resource at the App Store? 
Could you also document how to consume those created Apps?
Thanks in advance,

**Comment by ahemaid** [comment]
Hi @jfernandezsqs, 

Yes, the documentaion of the appStore is started, from time to time, it will be updated. 
Please find the link [https://international-data-spaces-association.github.io/IDS-AppStore/deployment](here)

**Comment by jfernandezsqs** [comment]
Hi,
I have tried to execute the python scripts located at IDS-AppStore/scripts/tests and both of them gave errors.
In the deployment with maven the Tests are skipped.
I have configured the App Store config.json to use a FH provided certificate and set the connector deploy mode to Productive.
>sqs@Virtual-Ubuntu-20:/IDS-AppStore/scripts/tests$ python3 contract_negotiation_allow_access.py 
Traceback (most recent call last):
  File "contract_negotiation_allow_access.py", line 4, in <module>
    import tqdm
ModuleNotFoundError: No module named 'tqdm'

>sqs@Virtual-Ubuntu-20:/IDS-AppStore/scripts/tests$ python3 contract_negotiation_allow_access.py 
{'error': 'Not Found',
 'path': '/api/ids/description',
 'status': 404,
 'timestamp': '2022-01-13T14:42:54.796+00:00'}
Traceback (most recent call last):
  File "contract_negotiation_allow_access.py", line 263, in <module>
    obj = offer["ids:contractOffer"][0]["ids:permission"][0]
KeyError: 'ids:contractOffer'
sqs@Virtual-Ubuntu-20:/IDS-AppStore/scripts/tests$ python3 contract_negotiation_allow_access.py 
Traceback (most recent call last):
  File "contract_negotiation_allow_access.py", line 4, in <module>
    import tqdm
ModuleNotFoundError: No module named 'tqdm'

>sqs@Virtual-Ubuntu-20:/IDS-AppStore/scripts/tests$ python3 contract_negotiation_allow_access.py 
{'error': 'Not Found',
 'path': '/api/ids/description',
 'status': 404,
 'timestamp': '2022-01-13T14:54:07.901+00:00'}
Traceback (most recent call last):
  File "contract_negotiation_allow_access.py", line 263, in <module>
    obj = offer["ids:contractOffer"][0]["ids:permission"][0]
KeyError: 'ids:contractOffer'

>sqs@Virtual-Ubuntu-20:/IDS-AppStore/scripts/tests$ python3 create_resources_and_upload_app_local.py 
Traceback (most recent call last):
  File "create_resources_and_upload_app_local.py", line 77, in post_request_check_response
    response_tmp.raise_for_status()
  File "/usr/lib/python3/dist-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error:  for url: http://localhost:8080/api/catalogs
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "create_resources_and_upload_app_local.py", line 477, in <module>
    catalog = create_catalog()
  File "create_resources_and_upload_app_local.py", line 121, in create_catalog
    loc = post_request_check_response(f"{combined_host}/api/catalogs", json)
  File "create_resources_and_upload_app_local.py", line 92, in post_request_check_response
    pprint(f"HTTP ERROR OCCURED: {http_error}")
TypeError: 'module' object is not callable

**Comment by ahemaid** [comment]
Could you please let us know if you still have those problems. I think we have sent you the updated version of the documentation?
Hope that helps you. Also please use the latest version of appstore and check if you still measure some issues.

**Comment by domreuter** [comment]
@jfernandezsqs The scripts need pythons tqdm module to be installed within your python environment.

This can be done via pip (pythons package manager):

`pip install tqdm`

Then give it another try ;-)

**Comment by jfernandezsqs** [comment]
After the follow up meeting, we were able to fix the issue with the python script `create_resources_and_upload_app.py` and we were able of uploading the app.
It is necessary to update the documentation for the configuration of the Python script so that an end user knows exactly the configuration required for an example set-up.
I close this issue because an update of the App Store documentation is coming and the issue #47 has been fixed.
